src/plugins/intel_cpu/tests/unit/snippets_transformations/aarch64/lowered/gemm_blocking.cpp

```diff
@@ -0,0 +1,209 @@
+    const ov::PartialShape input_shape_a{1, 16, m, k};
+    const ov::PartialShape input_shape_b{1, 16, k, n};
+    const auto precision = ov::element::f32;
+
+    {
+        auto data_a = linear_ir->push_node<ov::opset10::Parameter>(precision, input_shape_a);
+        auto data_b = linear_ir->push_node<ov::opset10::Parameter>(precision, input_shape_b);
+
+        auto gemm = linear_ir->push_node<aarch64::GemmCPU>(data_a.second, data_b.second,
+                                                           PortDescriptor{}, PortDescriptor{}, PortDescriptor{});
+        init_expr_descriptors(*gemm.first);
+        auto result = linear_ir->push_node<ov::opset10::Result>(gemm.second);
+    }
+    {
+        auto data_a = linear_ir_ref->push_node<ov::opset10::Parameter>(precision, input_shape_a);
+        auto data_b = linear_ir_ref->push_node<ov::opset10::Parameter>(precision, input_shape_b);
+
+        auto gemm = linear_ir_ref->push_node<aarch64::GemmCPU>(data_a.second, data_b.second,
+                                                               PortDescriptor{}, PortDescriptor{}, PortDescriptor{});
+        const auto& gemm_expr = *gemm.first;
+        init_expr_descriptors(gemm_expr, {{m_blk, full_dim}, {full_dim, n_blk}, {m_blk, n_blk}});
+        create_gemm_loop_infos(linear_ir_ref, gemm_expr, m, m_blk, 0, 0, n, n_blk);
+        gemm_expr->set_loop_ids({1, 0});
+        auto result = linear_ir_ref->push_node<ov::opset10::Result>(gemm.second);
+    }
+}
+
+TEST_F(GemmCPUBlockingTest, BlockingIsNotNeeded) {
+    const auto precision = ov::element::f32;
+    const ov::Dimension::value_type n = 64;
+    const ov::Dimension::value_type m = 32;
+    const ov::Dimension::value_type k = 16;
+    const ov::PartialShape input_shape_a{1, 16, m, k};
+    const ov::PartialShape input_shape_b{1, 16, k, n};
+
+    {
+        auto data_a = linear_ir->push_node<ov::opset10::Parameter>(precision, input_shape_a);
+        auto data_b = linear_ir->push_node<ov::opset10::Parameter>(precision, input_shape_b);
+        auto gemm = linear_ir->push_node<aarch64::GemmCPU>(data_a.second, data_b.second,
+                                                           PortDescriptor{}, PortDescriptor{}, PortDescriptor{});
+        init_expr_descriptors(*gemm.first);
+        auto result = linear_ir->push_node<ov::opset10::Result>(gemm.second);
+    }
+    {
+        auto data_a = linear_ir_ref->push_node<ov::opset10::Parameter>(precision, input_shape_a);
+        auto data_b = linear_ir_ref->push_node<ov::opset10::Parameter>(precision, input_shape_b);
+        auto gemm = linear_ir_ref->push_node<aarch64::GemmCPU>(data_a.second, data_b.second,
+                                                               PortDescriptor{}, PortDescriptor{}, PortDescriptor{});
+        const auto full_subtensor = VectorDims(2, ov::snippets::utils::get_full_dim_value());
+        init_expr_descriptors(*gemm.first, std::vector<VectorDims>(3, full_subtensor));
```

Comment:
If subtensors vec is not set, `init_expr_descriptors` will create full dim subtensors automatically, so we can omit it.
```suggestion
        init_expr_descriptors(*gemm.first);
```

Reply:
Applied